crates/bevy_ecs/src/world/mod.rs

```diff
@@ -3217,6 +3223,15 @@ impl World {
 }
 
 impl World {
+    /// Returns true if the type id is used internally, components and resources used internally
+    /// most likely should not be affected by things like saving and loading
+    pub fn is_internal_type(type_id: TypeId) -> bool {
+        if type_id == TypeId::of::<DefaultQueryFilters>() {
+            return true;
+        }
+        false
+    }
```

Comment:
I feel like this could be a little misleading. Users may expect this to return `true` for all internal-ish types. In their minds this might include computer components like `ComputedVisibility`, observer data, or other engine-specific types.

So rather than introducing this method now with only a single type actually included, what if we simply hardcode this logic in `be y_scene` where it seems to be used?